tests/callbacks/test_early_stopping.py

```diff
@@ -205,3 +208,94 @@ def validation_epoch_end(self, outputs):
     )
     trainer.fit(model)
     assert trainer.current_epoch >= 5, 'early_stopping failed'
+
+
+@pytest.mark.parametrize('step_freeze', [5, 7])
+@pytest.mark.parametrize('min_steps', [100] + list(range(4, 6)))
+@pytest.mark.parametrize('min_epochs', list(range(2, 3)))
+def test_min_steps_override_early_stopping_functionality(tmpdir, step_freeze, min_steps, min_epochs):
```

Comment:
please note that the longest test here is 0.93s which is significant considering this is one out of 1000+ tests.

Reply:
You are right. Let me reduce it :)

Reply:
yeah great, I think the current choice instead of the cartesian product is much better!